contracts/src/core/libraries/v3/config-registry/ConfigRegistryLib.sol

```diff
@@ -5,84 +5,168 @@ import {ConfigRegistryStorage as S} from "src/core/libraries/v3/config-registry/
     }
 
-    function getActivationKeyBytes(bytes32 nameDigest, uint256 index) internal view returns (uint256) {
-        return S.layout().bytesConfig.values[nameDigest][index].activationKey;
+    /// @notice Gets the activation block number at a specific index for a block number-based configuration
+    /// @param nameDigest The hash of the configuration name
+    /// @param index The index of the checkpoint to retrieve
+    /// @return The activation block number at the specified index
+    function getActivationKeyBlockNumber(bytes32 nameDigest, uint256 index) internal view returns (uint256) {
+        return S.layout().BlockNumberCfg.values[nameDigest][index].activationBlock;
     }
 
-    function getCheckpointBytes32(bytes32 nameDigest, uint256 index)
+    /// @notice Gets the full checkpoint at a specific index for a timestamp-based configuration
+    /// @param nameDigest The hash of the configuration name
+    /// @param index The index of the checkpoint to retrieve
+    /// @return The TimeStampCheckpoint containing both value and activation timestamp
+    function getCheckpointTimeStamp(bytes32 nameDigest, uint256 index)
         internal
         view
-        returns (T.Bytes32Checkpoint memory)
+        returns (T.TimeStampCheckpoint memory)
     {
-        return S.layout().bytes32Config.values[nameDigest][index];
+        return S.layout().TimeStampCfg.values[nameDigest][index];
     }
 
-    function getCheckpointBytes(bytes32 nameDigest, uint256 index) internal view returns (T.BytesCheckpoint memory) {
-        return S.layout().bytesConfig.values[nameDigest][index];
+    /// @notice Gets the full checkpoint at a specific index for a block number-based configuration
+    /// @param nameDigest The hash of the configuration name
+    /// @param index The index of the checkpoint to retrieve
+    /// @return The BlockNumberCheckpoint containing both value and activation block number
+    function getCheckpointBlockNumber(bytes32 nameDigest, uint256 index)
+        internal
+        view
+        returns (T.BlockNumberCheckpoint memory)
+    {
+        return S.layout().BlockNumberCfg.values[nameDigest][index];
     }
 
-    function addConfigBytes32(bytes32 nameDigest, uint256 activationKey, bytes32 value) internal {
-        T.Bytes32Cfg storage cfg = S.layout().bytes32Config;
+    /// @notice Adds a new timestamp-based configuration checkpoint
+    /// @param nameDigest The hash of the configuration name
+    /// @param activationTS The activation timestamp (must be > last activation timestamp for this config)
+    /// @param value The bytes configuration value
+    /// @dev For the first checkpoint, activationTS must be >= block.timestamp
+    /// @dev Subsequent checkpoints must have strictly increasing activation timestamps
+    function addConfigTimeStamp(bytes32 nameDigest, uint256 activationTS, bytes memory value) internal {
+        T.TimeStampCfg storage cfg = S.layout().TimeStampCfg;
```

Comment:
nit: throughout the codebase, we use write `timestamp` as a single word (even in your docs for these contracts). Therefore it seems odd to camelCase it as `TimeStamp`, as if it were two words: "time stamp"